### PR TITLE
feature/demoeng-752 (create demo service)

### PR DESCRIPTION
**Pull request by lvaladares**
## Why was this PR necessary?

https://issues.corp.twilio.com/browse/DEMOENG-752

## What does it do?
- Demo service as serverless nestjs service with lots of bells and whistles (prisma, swagger, class-validator, class-transformer)

## How to test?
- Add .env.local with your e2e mysql endpoint
- Then:
- ```npm run test:e2e```

### CODE REVIEW

File: src/demo/demo.controller.spec.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    demoService = module.get<DemoService>(DemoService);
+  });
+
+  it('should be defined', () => {
+    expect(sut).toBeDefined();
+  });
+
+  it('should call demoService with correct payload, return correct dto', async () => {
+    // const postDemoStub: PostDemoDto = {
+    //   name: '',
```
Comment (mandrioli-twilio) [inline src/demo/demo.controller.spec.ts]: Remove comments or fix test

File: src/index.ts [hunk 2]
```diff
@@ -1,26 +1,102 @@
...
+import { AppModule } from './app.module';
 
 let cachedServer: Server;
+let cachedLoggers: any;
+
+const setupSwagger = (app: INestApplication) => {
+  const options = new DocumentBuilder()
+    .setTitle('TSDaaS Demo Service')
+    .setVersion('1.0.0')
+    .addServer('http://localhost:3000/dev')
```
Comment (mandrioli-twilio) [inline src/index.ts]: Should be changed to allow the user to run on other ports

**Review by mandrioli-twilio — CHANGES_REQUESTED** [review]

File: src/index.ts (hunk 2 above)
Comment (mandrioli-twilio) [reply src/index.ts]: Also on src/main.ts file

**Review by vmiguelsilva — APPROVED** [review]

File: src/demo/demo.controller.spec.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { DemoController } from './demo.controller';
+import { DemoService } from './demo.service';
+import { Demo, PostDemoDto } from './model/demo.model';
+
+describe('DemoController', () => {
```
Comment (ghost) [inline src/demo/demo.controller.spec.ts:4]: Add more tests for this controller

File: src/index.ts [hunk 4]
```diff
@@ -1,26 +1,102 @@
...
+    .addServer('https://lvaladares.sa.ngrok.io/dev')
+    .build();
+  const document = SwaggerModule.createDocument(app, options);
+  SwaggerModule.setup('api', app, document);
+};
+
+const createLoggers = (context: Context) => {
```
Comment (ghost) [inline src/index.ts]: Extract function to a separate file

File: README.md [hunk 5]
```diff
@@ -114,3 +114,9 @@ $ npx ts-node src/swagger.ts
 Then browse http://localhost:3001/api
 
 **This function is for development.** If you want to use production, change package.json dependencies and serverless.yml.
+
```
Comment (ghost) [inline README.md:117]: Add instructions on how to run e2e tests on a containerized environment.

**Review by ghost — CHANGES_REQUESTED** [review]

File: test/2-property.e2e-spec.ts [hunk 6]
```diff
@@ -82,4 +82,19 @@ describe('PropertyController (e2e)', () => {
         expect(body).toBeTruthy();
       });
   });
+
```
Comment (ghost) [inline test/2-property.e2e-spec.ts:87]: If there's no more need for this test it should be deleted.

**Review by ghost — CHANGES_REQUESTED** [review]

File: src/index.ts (hunk 2 above)
Comment (lvaladares) [reply src/index.ts]: moved to .env file

File: src/demo/demo.controller.spec.ts (hunk 3 above)
Comment (lvaladares) [reply src/demo/demo.controller.spec.ts:4]: this is a default test file created from nest generator and is not being used in favor of e2e tests under the test/ folder

**Comment by lvaladares** [comment]
@aboscarino-twilio @mandrioli-twilio Reviewed

**Review by mandrioli-twilio — APPROVED** [review]
Looks good

**Pull request merged**